From c856be6bca6f963a9a2c91eda954ab687bc7e573 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Wed, 27 May 2015 08:37:55 +0200
Subject: [PATCH 419/506] multiboot1: never place modules in low memory.

While in theory permitted by the spec, modules rarely fit in low memory
anyway and not every kernel is able to handle modules in low memory anyway.
At least VMWare is known not to be able to handle modules at arbitrary
locations.
---
 grub-core/loader/multiboot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/grub-core/loader/multiboot.c b/grub-core/loader/multiboot.c
index 4b71f33..fd8f28e 100644
--- a/grub-core/loader/multiboot.c
+++ b/grub-core/loader/multiboot.c
@@ -333,6 +333,7 @@ grub_cmd_module (grub_command_t cmd __attribute__ ((unused)),
     return grub_errno;
 
 #ifndef GRUB_USE_MULTIBOOT2
+  lowest_addr = 0x100000;
   if (grub_multiboot_quirks & GRUB_MULTIBOOT_QUIRK_MODULES_AFTER_KERNEL)
     lowest_addr = ALIGN_UP (highest_load + 1048576, 4096);
 #endif
-- 
2.4.3

